### PR TITLE
rationalize exit codes in CSE

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: exit for every timeout, different exit code for every error.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: parts/k8s/kubernetescustomscript.sh [hunk 1]
```diff
@@ -24,12 +24,20 @@ ensureRunCommandCompleted()
 {
     echo "waiting for runcmd to finish"
     wait_for_file 900 1 /opt/azure/containers/runcmd.complete
+    if [ ! -f /opt/azure/containers/runcmd.complete ]; then
+        echo "Timeout waiting for runcmd to complete"
```
Comment (jackfrancis) [inline parts/k8s/kubernetescustomscript.sh]: can you add a mention of cloud-init here? maybe "Timeout waiting for cloud-init runcmd to complete"

**Comment by CecileRobertMichon** [comment]
> should we add comments at the top of this file that document the exit codes and what they indicate?


done

File: parts/k8s/kubernetescustomscript.sh [hunk 2]
```diff
@@ -1,5 +1,17 @@
 #!/bin/bash
 
+# This script runs on every Kubernetes VM
+# Exit codes represent the following:
+# | exit code number | meaning |
```
Comment (jackfrancis) [inline parts/k8s/kubernetescustomscript.sh]: How about if we reserve 1-9 for "general" exit failures, and then a decimal range (e.g., 10-19) per application-specific error condition. So, for example, "Etcd data dir not found" would be 10; "Timeout waiting for etcd to be accessible" would be 11; and then "Timeout waiting for docker install to finish" would be 20; etc

Comment (jackfrancis) [reply parts/k8s/kubernetescustomscript.sh]: Sorry, 1 and 2 should be reserved:

http://www.tldp.org/LDP/abs/html/exitcodes.html

Comment (jackfrancis) [reply parts/k8s/kubernetescustomscript.sh]: So, 3-9 for general errors:

- a service could not be enabled by systemctl
- a service could not be started by systemctl
- timeout waiting for cloud-init runcmd to complete
- timeout waiting for a file (ideally we'd have separate error messages per file we are waiting for, as they indicate a particular serial dependency)

Comment (CecileRobertMichon) [reply parts/k8s/kubernetescustomscript.sh]: updated them, is it what you had in mind?

**Pull request merged**